### ISSUE TITLE
The code uses more GPU memory with Multi-scale Vision Transformers

### ISSUE THREAD

**Issue by JunweiLiang** [facebookresearch/bitsandbytes#16]
Hi, 

Thanks for the great work! I'm currently trying to apply your code to vision transformers, specifically, on this code base:
https://github.com/facebookresearch/SlowFast/tree/main/projects/mvit
When using torch.optim.SGD(momentum=0.9), the code consumes 9221MiB GPU memory during training. After changing it to use bnb.optim.SGD8bit() with the same arguments, it consumes even a bit more GPU memory of 9235MiB. Do you have any idea why this would happen? Thank you! My CUDA version is 10.2 and torch version is 1.9.1.

Best,
Junwei

**Comment by TimDettmers** [comment]
Hi Junwei, how many parameters does the model have? From the paper, it seems some models have about 50M parameters from which you would expect to save about ~150MB of memory. If you look at the memory of `nvidia-smi`, then you see maximum allocated memory which is not necessarily equal to the actual memory usage. It is closer to peak usage. 

Do you see this excessive memory consumption with you start the model with these setups with different random seeds?

**Comment by JunweiLiang** [comment]
Hi Tim, 

Thanks for replying. Based on the [MViT paper](https://arxiv.org/pdf/2104.11227.pdf)(Table 5), the one I'm using (MViT-B_32x3) has 36.8 million parameters. I'm not sure why this model would takes 9GB GPU memory in peak usage even with batch_size=1 per GPU in the first place.
I changed the random seed and the GPU memory consumption with SGD8bit did not change. I think if in theory we are looking at only 36.8*3 MB memory saving, it would not be noticeable from `nvidia-smi`.
